src/test/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilderUnitTest.java

```diff
@@ -347,6 +347,25 @@ public void testIndexingOnTheFly() {
     }
   }
 
+  @Test
+  public void testIndexingOnTheFlyForPathStream() throws IOException {
+    final VariantContextWriterBuilder builder = new VariantContextWriterBuilder()
+            .setReferenceDictionary(dictionary)
+            .setOption(Options.INDEX_ON_THE_FLY);
+
+    try (FileSystem fs = Jimfs.newFileSystem("test", Configuration.unix())) {
+      final Path vcfPath = IOUtil.getPath("/dev/null");
+      final Path vcfIdxPath = Tribble.indexPath(vcfPath);
+      // We explicitly setOutputFileType here to mimic gatk behavior where it pretends files that have no extension
+      try (final VariantContextWriter writer = builder.setOutputPath(vcfPath).setOutputFileType(OutputType.VCF_STREAM).build()) {
```

Comment:
will this also work if you do not set the output type to a stream?

Reply:
No it won't. In that case it will end up in the same place where we are as of right now, where an output stream location will end up "UNSPECIFIED" and throw an exception unless the user explicitly defines what type of output it is.